File: editor/en/[1347]Minimum Number of Steps to Make Two Strings Anagram.java

//leetcode submit region begin(Prohibit modification and deletion)
class Solution {
    public int minSteps(String s, String t) {
        int[] countS = getCount(s);
        int[] countT = getCount(t);
        int count = 0;

        for (int i = 0; i < countS.length; i++) {
            if (countT[i] < countS[i]) {
                count += countS[i] - countT[i];
            }
        }

        return count;
    }

    private int[] getCount(String s) {
        int[] arr = new int[26];
        for (char c : s.toCharArray()) {
            arr[c - 'a']++;
        }

        return arr;
    }
}
//leetcode submit region end(Prohibit modification and deletion)

File: editor/en/[1347]Minimum Number of Steps to Make Two Strings Anagram.js
//leetcode submit region begin(Prohibit modification and deletion)
/**
 * @param {string} s
 * @param {string} t
 * @return {number}
 */
var minSteps = function (s, t) {
    const getCount = (s) => {
        let arr = new Array(26).fill(0);
        for (const c of s) {
            arr[c.charCodeAt(0) - 'a'.charCodeAt(0)]++;
        }

        return arr;
    }

    let countS = getCount(s);
    let countT = getCount(t);
    let count = 0;

    for (let i = 0; i < countS.length; i++) {
        if (countS[i] > countT[i]) {
            count += countS[i] - countT[i];
        }
    }

    return count;
};
//leetcode submit region end(Prohibit modification and deletion)
